Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -508,15 +523,30 @@ internal async void ClearAllPoolsAsync()
             }
         }
 
-        public void SetMaxPoolSize(int size)
+        public void SetMaxPoolSize(int newMaxPoolSize)
         {
-            _poolConfig.MaxPoolSize = size;
+            s_logger.Info($"SessionPool::SetMaxPoolSize({newMaxPoolSize})");
+            _poolConfig.MaxPoolSize = newMaxPoolSize;
             _configOverriden = true;
         }
 
-        public int GetMaxPoolSize()
+        public int GetMaxPoolSize() => _poolConfig.MaxPoolSize;
+
+        public int GetMinPoolSize()
+        {
+            return IsMultiplePoolsVersion()
```

Comment:
If we decide to throw an error for GetMinPoolSize, then maybe we should do the same for others method which have no meaning for the old pool like GetWaitForIdleSessionTimeout(), etc.

Reply:
fixed